src/accelerate/deepspeed_utils.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.nn as nn
+from dataclasses import dataclass, field
+
+from .state import is_apex_available
+
+@dataclass
+class DeepSpeedPlugin:
+
+    gradient_accumulation_steps: int = field(default=1, metadata={"help": "Number of steps to accumulate gradients before updating optimizer states"})
+    zero_stage: int = field(default=0, metadata={"help": "Possible options are 0,1,2"})
+    is_train_batch_min: str = field(default=True, metadata={"help": "If both train & eval dataloaders are specified, this will decide the train_batch_size"})
+
+    fp16: bool = field(repr=False, default=False, metadata={"help": "You need not define this here"})
+
+    def __post_init__(self):
+        self.ds_config = {
+            "train_batch_size": None,
+            "fp16": {"enabled": False if self.fp16 is None else self.fp16},
+            "gradient_accumulation_steps": self.gradient_accumulation_steps,
+            "zero_optimization": {"stage": self.zero_stage},
+            "steps_per_print": float("inf"), # this will stop deepspeed from logging @ stdout
+        }
+
+
+class DeepSpeedEngineWrapper(nn.Module):
```

Comment:
This is copied from `deepspeed`? Could we subclass their class then and only change the method that requires it?

Reply:
Yes, we can do that I think. Then  we won't need `deepspeed.initialize()` since deepspeed.initialize() is doing just sending everything to `DeepSpeedEngine` (see [this](https://github.com/microsoft/DeepSpeed/blob/46f4573b1a8a9cda2b45f4de4e90b631cee3f80b/deepspeed/__init__.py#L119)).